frontend/src/lib/validation.ts

```diff
@@ -211,3 +211,19 @@ export function validateBareMetalOSImageURL(value: string, t: TFunction) {
 
     return undefined
 }
+
+export function validateURL(url: string, t: TFunction) {
+    const regex_injected_spaces = /(\S)*(\s)(\S)*/
+    let towerUrl = null
+
+    try {
+        towerUrl = new URL(url)
+    } catch (err) {
+        return t('validate.ansible.url.not.valid')
+    }
+
+    if (towerUrl.protocol.toString() === 'https:' || towerUrl.protocol.toString() === 'http:')
+        if (!regex_injected_spaces.test(url)) return undefined
+
+    return t('validate.ansible.url.not.valid')
+}
```

Comment:
@KevinFCormier 
So I've kept some regex because ```new URL``` does not error when it encounters spaces within the url string.  It also accepts basically any string for a protocol, as long as it's ended correctly.
This new regex picks up on inner spaces, trailing, and leading spaces-making me wonder if the trimming method I've updated is still necessary?  I'm also explicitly checking the protocol. Thoughts?

Reply:
Hey Randy,

Thanks for looking at this more.

1. I think my advice was bad. The URL parsing doesn't seem to do much validation at all. In addition to spaces, I can put lots of invalid characters in there, like underscores in the domain part, punctuation, etc. I guess we could go back to a regex, or maybe @jamestalton has a suggestion? There are some packages available like https://www.npmjs.com/package/validator that give an isURL function with a bunch of options.
1. On the question of whether the trimming method is needed - depends whether you want to mark spaces or trailing / as invalid, or whether you want to automatically remove them.